src/service/resource.js

```diff
@@ -560,6 +566,32 @@ export class Resource {
   }
 
   /**
+   * @return {!Promise} resolves when underlying element is built and within the
+   *    range specified by implementation's renderOutsideViewport
+   */
+  whenWithinRenderOutsideViewport() {
+    if (this.renderOutsideViewportPromise_) {
+      return this.renderOutsideViewportPromise_;
+    }
+    return this.renderOutsideViewportPromise_ = new Promise(resolver => {
+      this.renderOutsideViewportResolve_ = resolver;
+    });
+  }
+
+  /**
+   * @private resolves render outside viewport promise if one was created via
+   *    whenWithinRenderOutsideViewport.
+   */
+  resolveRenderOutsideViewport_() {
```

Comment:
Added test coverage and tested locally with multiple slots and did not see any race condition issues.  We can control the experiment ramp up server side and so can do gradually and measure the effect.